nifi-nar-bundles/nifi-cybersecurity-bundle/nifi-cybersecurity-processors/src/main/java/org/apache/nifi/processors/cybersecurity/CompareFuzzyHash.java

```diff
@@ -0,0 +1,358 @@
+            .required(true)
+            .allowableValues(singleMatch,multiMatch)
+            .defaultValue(singleMatch.getValue())
+            .build();
+
+    public static final Relationship REL_MATCH = new Relationship.Builder()
+            .name("Matched")
+            .description("Any FlowFile that is successfully matched to an existing hash will be sent to this Relationship.")
+            .build();
+
+    public static final Relationship REL_NON_MATCH = new Relationship.Builder()
+            .name("non-match")
+            .description("Any FlowFile that cannot be matched to an existing hash will be sent to this Relationship.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("Failure")
+            .description("Any FlowFile that cannot be matched, e.g. (lacks the attribute) will be sent to this Relationship.")
+            .build();
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(HASH_LIST_FILE);
+        // As mentioned above, add the PropertyDescriptor HASH_ALGORITHM and ATTRIBUTE_NAME from parent class
+        descriptors.add(HASH_ALGORITHM);
+        descriptors.add(ATTRIBUTE_NAME);
+        descriptors.add(MATCH_THRESHOLD);
+        descriptors.add(MATCHING_MODE);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_MATCH);
+        relationships.add(REL_NON_MATCH);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+        String algorithm = context.getProperty(HASH_ALGORITHM).getValue();
+
+        String inputHash = flowFile.getAttribute(context.getProperty(ATTRIBUTE_NAME).getValue());
+
+        if (inputHash == null) {
+            getLogger().info("FlowFile {} lacks the required '{}' attribute, routing to failure.",
+                    new Object[]{flowFile, context.getProperty(ATTRIBUTE_NAME).getValue() });
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        Digest inputDigest = null;
+        SpamSum spamSum = null;
```

Comment:
In fact, thinking about exposing the common interface would allow for a method `boolean matchExceedsThreshold(double match)` which avoids the "some hashes use higher numbers to indicate similarity, some use lower" issue as well.